pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -1176,9 +1189,17 @@ func Test_server_Insert(t *testing.T) {
 		fields     fields
 		want       want
 		checkFunc  func(want, *payload.Object_Location, error) error
-		beforeFunc func(args)
+		beforeFunc func(*testing.T, *test)
```

Comment:
I think passing the test struct is too much and the fields struct is better.
And I think returning service.NGT is better than setting the fields struct.

Reply:
> passing the test struct is too much and the fields struct is better.
@vankichi what do you think about this? Both (test or fields struct) is fine for me.

> returning service.NGT is better than setting the fields struct.
Why I design it like this is because in case the test case needs to set the NGT object directly instead of initialize it via config. 
What do you think about this?

Reply:
IMO, it is better to get only `fields.svcCfg` and `fields.svcOpts` in this case.
If this function is defined as a global default function (out of `func Test_xxx() {}`), it can be.

How about these flow?
it does not have to read a test struct.
```go
// define
type fields {
  ...
  ngt ngt
  ...
}
// test loop
for _, test := range tests {
    ...
    test.fileds.ngt, err := test.beforeFunc(fields.svcCgf, fieds.svcOpts...)
    if err != nil {
        t.Fatal(err)
    }
    ...
}
```

Reply:
Thank you for your opinion.

Personally I think returning value in `beforeFunc()` is not good as I suppose it to set the prerequisites of the test case (like setting env variable or like I use `beforeFunc()` to insert a duplicated vector before the test case run. e.g. `case 1.1: Insert duplicated request fail when SkipStrictExistCheck is off (duplicated ID)`),.

If you don't like the current design, I recommended to create a separate func `fieldFunc()` or `argsFunc()` to set the field or args. 

```go
type fields { 
   ... 
   ngt ngt 
   ... 
}

// test loop
for _, test := range tests { 
   ... 
   test.fileds.ngt, err := test.FieldFunc(fields.svcCgf, fieds.svcOpts...) 
   if err != nil { t.Fatal(err) } 
   .. 
}
```

FYI, why I design like this is for flexibility. User can easily intercept their logic before the test case run (like initializing the NGT object in this case)

Reply:
In the flow, I think that it is not necessary for ngt in fields struct.
However, the basic idea sounds very good.

Reply:
@kmrmt do you mean my current implementation (on this branch) is good enough for you?

Reply:
Unfortunately, I do not agree about this implementation yet.

Reply:
Why do we need to use beforeFunc for creating NGT server?
It's simple to create in each test loop without beforeFunc.

Reply:
fixed